Assets/Scripts/Minigame/MinigameBaseClass.cs

```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+
+namespace Minigame
+{
+    public class MinigameBaseClass : MonoBehaviour
+    {
+        public bool IsDone {  get; protected set; } = false;
+
+        public virtual void WakeUp()

```

Comment:
using empty virtual methods is okay, but serves no "real" purpose. Usually you'd go for an abstract method instead: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/abstract